modules/carto/src/config.js

```diff
@@ -0,0 +1,33 @@
+const defaultCredentials = {
+  username: 'public',
+  apiKey: 'default_public',
+  region: 'us',
+  // Set to null to guess from mapsUrl attribute. Other values are 'v1' or 'v2'
+  mapsVersion: null,
+  // SQL API URL
+  sqlUrl: 'https://{user}.carto.com/api/v2/sql',
+  // Maps API URL - v1 endpoint
+  mapsUrl: 'https://{user}.carto.com/api/v1/map'
+};
+
+let credentials = defaultCredentials;
+
+export function setDefaultCredentials(opts) {
+  credentials = {
+    ...credentials,
+    ...opts
+  };
+}
+
+export function getDefaultCredentials() {
+  return credentials;
+}
+
+export function getMapsVersion(creds) {
```

Comment:
I see clearly the point for the getDefaultCredentials / setDefaultCredentials.... what's the use case for `getMapsVersion`?

Reply:
It's used from  CartoSQLLayer to get the mapsVersion to generate the mapConfig of v1 or v2.

Also used at maps-api-client to parse the response of mapsAPI v1 or v2

Reply:
Ok, I see